src/Service/Resolvers/Sql Query Structures/BaseSqlQueryStructure.cs

```diff
@@ -506,5 +492,51 @@ public void ProcessOdataClause(FilterClause odataClause)
         {
             return filterClause.Expression.Accept<string>(visitor);
         }
+
+        /// <summary>
+        /// Gets the value of the parameter cast as the system type
+        /// </summary>
+        protected object GetParamAsSystemType(string param, string paramName, Type systemType)
+        {
+            try
+            {
+                return ParseParamAsSystemType(param, systemType);
+            }
+            catch (Exception e)
+            {
+                if (e is FormatException ||
+                    e is ArgumentNullException ||
+                    e is OverflowException)
+                {
+                    string errorMessage;
+                    if (MetadataProvider.IsDevelopmentMode())
+                    {
+                        if (MetadataProvider.EntityToDatabaseObject[EntityName].SourceType is SourceType.StoredProcedure)
+                        {
+                            errorMessage = $@"Parameter ""{param}"" cannot be resolved as stored procedure parameter ""{paramName}"" " +
+                                    $@"with type ""{systemType.Name}"".";
+                        }
+                        else
+                        {
+                            errorMessage = $"Parameter \"{param}\" cannot be resolved as column \"{paramName}\" " +
+                                    $"with type \"{systemType.Name}\".";
+                        }
+                    }
+                    else
+                    {
+                        errorMessage = $"Invalid value provided for field: {paramName}";
```

Comment:
Is this going to be the mapped name or backing column name? I think it should be the mapped name - because that is the exposed field